### PR TITLE
add unittest for RequiredEnv func

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

1. add unittest for RequiredEnv func

### CODE REVIEW

File: pkg/tmpl/context_funcs_test.go [hunk 1]
```diff
@@ -221,3 +223,31 @@ func TestRequired(t *testing.T) {
...
+	envVal, err = RequiredEnv(envKey)
+
+	require.NotNilf(t, err, "Expected error to be returned when environment variable %s is set to an empty string", envKey)
+	require.Emptyf(t, envVal, "Expected empty string to be returned when environment variable %s is set to an empty string", envKey)
+
+	// test that the environment variable is set to a non-empty string
+	os.Setenv(envKey, "helmfile")
+
+	// Unset the environment variable
+	defer os.Unsetenv("HelmFile")
```
Comment (itscaro) [inline pkg/tmpl/context_funcs_test.go:248]:
```suggestion
	defer os.Unsetenv(envKey)
```

File: pkg/tmpl/context_funcs_test.go [hunk 2]
```diff
@@ -221,3 +223,31 @@ func TestRequired(t *testing.T) {
...
+
+	// test that the environment variable is set to an empty string
+	os.Setenv(envKey, "")
+	envVal, err = RequiredEnv(envKey)
+
+	require.NotNilf(t, err, "Expected error to be returned when environment variable %s is set to an empty string", envKey)
+	require.Emptyf(t, envVal, "Expected empty string to be returned when environment variable %s is set to an empty string", envKey)
+
+	// test that the environment variable is set to a non-empty string
+	os.Setenv(envKey, "helmfile")
```
Comment (itscaro) [inline pkg/tmpl/context_funcs_test.go:245]:
```suggestion
	expected := "helmfile"
	os.Setenv(envKey, expected)
```

File: pkg/tmpl/context_funcs_test.go [hunk 3]
```diff
@@ -221,3 +223,31 @@ func TestRequired(t *testing.T) {
...
+
+	// Unset the environment variable
+	defer os.Unsetenv("HelmFile")
+	envVal, err = RequiredEnv(envKey)
+	require.Nilf(t, err, "Expected no error to be returned when environment variable %s is set to a non-empty string", envKey)
+	require.Equalf(t, "helmfile", envVal, "Expected environment variable %s to be returned when environment variable %s is set to a non-empty string", envVal, envKey)
```
Comment (itscaro) [inline pkg/tmpl/context_funcs_test.go:251]:
```suggestion
	require.Equalf(t, expected, envVal, "Expected environment variable %s to be returned when environment variable %s is set to a non-empty string", envVal, envKey)
```

File: pkg/tmpl/context_funcs_test.go [hunk 4]
```diff
@@ -221,3 +223,32 @@ func TestRequired(t *testing.T) {
...
+
+// TestRequiredEnv tests that RequiredEnv returns an error if the environment variable is not set.
+func TestRequiredEnv(t *testing.T) {
+
+	// test that the environment variable is not set
+	envKey := "HelmFile"
+	envVal, err := RequiredEnv(envKey)
+
+	require.NotNilf(t, err, "Expected error to be returned when environment variable %s is not set", envKey)
+	require.Emptyf(t, envVal, "Expected empty string to be returned when environment variable %s is not set", envKey)
```
Comment (mumoshu) [inline pkg/tmpl/context_funcs_test.go:235]: Nit but AFAIK any error stops template execution when it retuned an non-nil error it won't matter match what's returned when it errored! But it's also true there might be no bad side-effect having this covered in the test.

**Review by mumoshu — APPROVED** [review]
Thank you so much for your continued effort 🙏

**Pull request merged**